### PR TITLE
[main] Remove unneeded patches

### PR DESCRIPTION
**Pull request by matzew**
A bit of tech dept. removal of unneeded patches, thanks to PR #1459

### CODE REVIEW

File: knative-operator/deploy/resources/knativekafka/channel/2-channel-post-install.yaml [hunk 1]
```diff
@@ -109,12 +109,13 @@ roleRef:
 apiVersion: batch/v1
 kind: Job
 metadata:
-  generateName: storage-version-migration-kafka-channel-
+  generateName: storage-version-migration-eventing-kafka-channel-
```
Comment (pierDipi) [inline knative-operator/deploy/resources/knativekafka/channel/2-channel-post-install.yaml:112]: This name needs to match up with what we pass to the CSV.

https://github.com/openshift-knative/serverless-operator/blob/3f8441fda1b6092a49247f1aab328f3815df43c6/hack/generate/csv.sh#L77

/hold

Comment (pierDipi) [reply knative-operator/deploy/resources/knativekafka/channel/2-channel-post-install.yaml:112]:
```
kafka_image "storage-version-migration-eventing-kafka-channel-__migrate" "${eventing_kafka}-storage-version-migration" 
```
should work

**Comment by matzew** [comment]
what is

```
run golangci-lint
  Running [/home/runner/golangci-lint-1.43.0-linux-amd64/golangci-lint run --out-format=github-actions --path-prefix=./src/github.com/openshift-knative/serverless-operator] in [/home/runner/work/serverless-operator/serverless-operator/src/github.com/openshift-knative/serverless-operator] ...
  panic: load embedded ruleguard rules: rules/rules.go:13: can't load fmt
  
  goroutine 1 [running]:
  github.com/go-critic/go-critic/checkers.init.9()
  	github.com/go-critic/go-critic@v0.6.1/checkers/checkers.go:58 +0x4b4
  
  Error: golangci-lint exit with code 2
  Ran golangci-lint in 15[90](https://github.com/openshift-knative/serverless-operator/runs/5831432049?check_suite_focus=true#step:7:90)9ms
```

**Comment by matzew** [comment]
/hold

let's NOT do that short before the release.....

**Comment by dsimansk** [comment]
@matzew that started to pop-up in the upstream lint checks as well. It should be fixed with newer golangcli-lint version.

**Comment by aliok** [comment]
Related: https://github.com/openshift-knative/serverless-operator/pull/1473

Let's merge this first and I will adapt the changes in #1473

**Comment by aliok** [comment]
/lgtm
/approve

Not touching the `/hold`.

**Comment by pierDipi** [comment]
@aliok @matzew this is a bug https://github.com/openshift-knative/serverless-operator/pull/1464#discussion_r842587841

**Comment by pierDipi** [comment]
actually, it is fixed, all good :+1:

**Pull request merged**